cascade/cascade.py

```diff
@@ -392,6 +410,50 @@ def _check_pull_output_overload(self, stderr: str) -> bool:
         """
         return any([x in stderr for x in _DOCKER_PULL_ERRORS])
 
+    def _get_singularity_pull_cmd(self, image: str) -> str:
+        """Get singularity pull command
+        :param str image: image to pull
+        :rtype: str
+        :return: pull command for the singularity image
+        """
+        # if we have a key_fingerprint we need to pull
+        # the key to our keyring
+        image_out_path = singularity_image_path_on_disk(image)
+        key_file_path = pathlib.Path(
+            singularity_image_name_to_key_file_name(image))
+        if image in _DIRECTDL_KEY_FINGERPRINT_DICT:
+            singularity_pull_cmd = ('singularity pull {} {}'
+                                    .format(image_out_path, image))
+            key_fingerprint = _DIRECTDL_KEY_FINGERPRINT_DICT[image]
+            if key_file_path.is_file():
+                key_import_cmd = ('singularity key import {}'
+                                  .format(key_file_path))
+                fingerprint_check_cmd = (
+                    'key_fingerprint=$({} | '.format(key_import_cmd) +
+                    'grep -o "fingerprint \\(\\S*\\)" | ' +
+                    'grep -o "\\S*$"); ' +
+                    'if [ $key_fingerprint != ' +
```

Comment:
Change `$key_fingerprint` to `${key_fingerprint^^}` and next line do `key_fingerprint.upper()` on the py var.